### PR TITLE
Detail Activity Dagger Modules

### PR DESCRIPTION
**Pull request by CebotariM**
Detail Activity Refactoring with Dagger

### CODE REVIEW

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/mapper/DataMapper.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            cityForecast.setCityName(city.getName());
+            WeatherData weatherDataForNow = weatherDataService.getUnique(cityForecast.getCityId(), currentTime);
+            cityForecast.setCurrentCityWeather(weatherDataForNow);
+            cityForecastList.add(cityForecast);
+        }
+
+        return cityForecastList;
+    }
+
+    private DailyForecast getDailyForecast(Long cityId, Long dt) {
```
Comment (mbohnearschi) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/mapper/DataMapper.java]: no more variable names as "dt". this decreases code readability.

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/mapper/DataMapper.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        dailyForecast.setWeatherDataList(weatherDataForDay);
+        dailyForecast.setCityId(cityId);
+        dailyForecast.setDayTempMax(getTempMax(weatherDataForDay));
+        dailyForecast.setDayTempMin(getTempMin(weatherDataForDay));
+
+        dailyForecast.setDate(TimeUtils.setLongToDate(dt));
+        return dailyForecast;
+    }
+
+    public List<DailyForecast> getDailyForecastList(Long cityId, Long dt, int numberOfDays) {
```
Comment (mbohnearschi) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/mapper/DataMapper.java]: "dt" - how am I supposed to know what this is?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java [hunk 3]
```diff
@@ -2,83 +2,41 @@
...
-        Long timestamp = bundle.getLong(TIMESTAMP_KEY);
-
-        DetailsPresenter detailsPresenter = new DetailsPresenter(WeatherApplication.getAppComponent().cityService(), WeatherApplication.getAppComponent().weatherDataService());
 
-        collapsingToolbarLayout.setTitle(detailsPresenter.getCityName(cityId));
-        collapsingToolbarLayout.setCollapsedTitleTextColor(Color.WHITE);
-        collapsingToolbarLayout.setExpandedTitleColor(Color.WHITE);
+        DaggerDetailComponent.builder()
+                .appComponent(WeatherApplication.getAppComponent())
+                .detailModule(new DetailModule(this, bundle.getLong(CITY_ID_KEY), bundle.getLong(TIMESTAMP_KEY)))
```
Comment (VdovinN) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java]: Guys, what do you think about it? Pass cityId and timestamp in setter in DetailActivity???

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    CollapsingToolbarLayout collapsingToolbarLayout;
+
+    private MainRecyclerAdapter mainRecyclerAdapter;
+    private Context context;
+    private DetailsPresenter detailsPresenter;
+    private View detailsView;
+
+    public DetailView(DetailActivity detailActivity,
+                      MainRecyclerAdapter mainRecyclerAdapter,
+                      DetailsPresenter detailsPresenter) {
```
Comment (VdovinN) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java]: Guys, are you satisfied that View's constructor has presenter?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java (hunk 3 above)
Comment (mbohnearschi) [reply app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java]: Take a look at how I start activities. Have a static "start" method that takes anything you need and then add to the bundle. The module only takes the activity and that's it.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java (hunk 4 above)
Comment (mbohnearschi) [reply app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java]: View must not know about the presenter. Only the Presenter knows about the view and the model.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        mainRecycler.setLayoutManager(new LinearLayoutManager(context));
+        mainRecycler.setVerticalFadingEdgeEnabled(true);
+        mainRecycler.setAdapter(new MainRecyclerAdapter(
+                context,
+                dailyForecastList,
+                weatherData
+        ));
+    }
+    private void historyClickHandler(){
+        Intent historyIntent = new Intent(context, HistoryActivity.class);
```
Comment (mbohnearschi) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java]: This should not be in the view, the click listener should come from presenter. And this should also be handled in the presenter.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/dagger/GridHistoryFragmentModule.java [hunk 6]
```diff
@@ -39,8 +40,8 @@ GridHistoryView provideGridHistoryView(GridHistoryFragment gridHistoryFragment,
 
     @Provides
     @GridHistoryFragmentScope
-    GridHistoryPresenter provideGridHistoryPresenter(CityService cityService, WeatherDataService weatherDataService, GridHistoryView gridHistoryView) {
-        return new GridHistoryPresenter(cityService, weatherDataService, gridHistoryView);
+    GridHistoryPresenter provideGridHistoryPresenter(DataMapper dataMapper, GridHistoryView gridHistoryView) {
```
Comment (mbohnearschi) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/dagger/GridHistoryFragmentModule.java:35]: There is no need for this "@Provides
     @GridHistoryFragmentScope
     GridHistoryFragment provideGridHistoryFragment() {
         return gridHistoryFragment;
     }"

You can just use the gridHistoryFragment which you already have as a private variable (GridHistoryView)

**Comment by MaximCeban** [comment]
@CebotariM  Could you please do not rebase commits until you get approved to merge, just to keep history of changes.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/dagger/GridHistoryFragmentModule.java (hunk 6 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/dagger/GridHistoryFragmentModule.java:35]: *Fixed

**Review by victorjimbei — APPROVED** [review]

**Pull request merged**